libs/coin-evm/src/api/explorer/ledger.ts

```diff
@@ -0,0 +1,131 @@
+import axios from "axios";
+import { getEnv } from "@ledgerhq/live-env";
+import { delay } from "@ledgerhq/live-promise";
+import { Operation } from "@ledgerhq/types-live";
+import { isNFTActive } from "@ledgerhq/coin-framework/nft/support";
+import { ExplorerApi, isLedgerExplorerConfig } from "./types";
+import { LedgerExplorerUsedIncorrectly } from "../../errors";
+import { LedgerExplorerOperation } from "../../types";
+import {
+  ledgerERC20EventToOperations,
+  ledgerERC721EventToOperations,
+  ledgerERC1155EventToOperations,
+  ledgerOperationToOperations,
+} from "../../adapters/index";
+
+export const BATCH_SIZE = 10_000;
+export const LEDGER_TIMEOUT = 200; // 200ms between 2 calls
+export const DEFAULT_RETRIES_API = 2;
+
+type OperationsRequestParams = {
+  explorerId: string;
+  address: string;
+  fromBlock?: number;
+};
+
+/**
+ * Request fetching all operations from an address
+ * and supporting pagination through tokens.
+ */
+export async function fetchPaginatedOpsWithRetries(
+  params: Required<OperationsRequestParams>,
+  paginationToken: string | null = null,
+  previousOperations: LedgerExplorerOperation[] = [],
+  retries = DEFAULT_RETRIES_API,
+): Promise<LedgerExplorerOperation[]> {
+  try {
+    const {
+      data: { data: operationsBatch, token },
+    } = await axios.request<{
+      data: LedgerExplorerOperation[];
+      token: string;
+    }>({
+      headers: { "X-Ledger-Client-Version": getEnv("LEDGER_CLIENT_VERSION") },
```

Comment:
(same comment here regarding the header)

Reply:
Answered in the first comment 👍